sycl/include/sycl/ext/intel/esimd/detail/intrin.hpp

```diff
@@ -64,8 +64,10 @@
 //
 template <typename T, int N, int M, int VStride, int Width, int Stride,
           int ParentWidth = 0>
-__ESIMD_INTRIN __ESIMD_DNS::vector_type_t<T, M>
-__esimd_rdregion(__ESIMD_DNS::vector_type_t<T, N> Input, uint16_t Offset);
+__ESIMD_INTRIN __ESIMD_INTRIN std::enable_if_t<M % Width == 0 && (Width > 0),
```

Comment:
Wouldn't it be better to check (Width > 0) before potential div-by-zero: (M % Width) ?